### PR TITLE
feat: add stop words / repository context support for C/C++

### PR DESCRIPTION
**Pull request by Squadrick**
Requires:
- https://github.com/tree-sitter/tree-sitter-c/pull/175
- https://github.com/tree-sitter/tree-sitter-cpp/pull/233

### CODE REVIEW

File: crates/tabby-common/assets/languages.toml
Comment (wsxiaoys) [inline crates/tabby-common/assets/languages.toml:95]: should be `//`?

File: crates/tabby-common/assets/languages.toml
Comment (wsxiaoys) [inline crates/tabby-common/assets/languages.toml:111]: ditto

File: crates/tabby-common/assets/languages.toml
Comment (Squadrick) [reply crates/tabby-common/assets/languages.toml]: Whoops, brain derp. Fixed.

File: crates/tabby-common/assets/languages.toml
Comment (Squadrick) [reply crates/tabby-common/assets/languages.toml]: fixed

**Comment by Squadrick** [comment]
@wsxiaoys The PRs to exposing `TAGS_QUERY` (above) have been merged, but there wasn't been a new release tag with those commits. I've raised a request for new tags with the maintainers of those repos. In the meanwhile, I can use cargo's `rev` property to pick the code, but imo, it's a bit ugly. 

So instead of:
```
tree-sitter-c = "0.20.3"
tree-sitter-cpp = "0.20.3"
```
we'll have:
```
tree-sitter-c = { git = "https://github.com/tree-sitter/tree-sitter-c/", rev = "4add3bb" }
tree-sitter-cpp = { git = "https://github.com/tree-sitter/tree-sitter-cpp", rev = "d153fe1" }
```

What do you think? Go with `rev` or wait for release tags?

**Comment by wsxiaoys** [comment]
Go with rev is ok - but you might also considering simply embeds the query scm in https://github.com/TabbyML/tabby/tree/main/crates/tabby-scheduler/queries

Like what we did for go / tsx

**Comment by Squadrick** [comment]
@wsxiaoys Since the queries were not modified, I went with the `rev` approach. If you would prefer that we stick to having the queries in-repo, let me know and I'll make that changes. 

Squashed and pushed the final commit, can you please take a look?

**Comment by wsxiaoys** [comment]
Could you add a few verification case (through screenshot) in treesitter's query playground" https://tree-sitter.github.io/tree-sitter/playground,  like what we did in https://github.com/TabbyML/tabby/pull/813#discussion_r1398541163

**Comment by Squadrick** [comment]
@wsxiaoys Sorry about the delay, here's the verification for C++:

Source code:
```c++
struct A {
	int a;
    void method();
};
class B {
	void method();
};
void B::method() {}
union C {
  A a; B b;
};
void printTest(int a, std::string b) {}
typedef struct Person {
	std::string s;
} p;
```

Query:
```
(struct_specifier name: (type_identifier) @name body:(_)) @definition.class
(union_specifier name: (type_identifier) @name) @definition.class
(function_declarator declarator: (identifier) @name) @definition.function
(function_declarator declarator: (field_identifier) @name) @definition.function
(function_declarator declarator: (qualified_identifier scope: (namespace_identifier) @local.scope name: (identifier) @name)) @definition.method
(type_definition declarator: (type_identifier) @name) @definition.type
(enum_specifier name: (type_identifier) @name) @definition.type
(class_specifier name: (type_identifier) @name) @definition.class
```

Image:
Screenshot 2023-12-15 at 1 42 53 PM

**Comment by Squadrick** [comment]
For C:
```c
struct A {
	int a;
    void method();
};
void printTest(int a, std::string b) {}
typedef struct Person {
	std::string s;
} p;
```

Query:
```
(struct_specifier name: (type_identifier) @name body:(_)) @definition.class
(declaration type: (union_specifier name: (type_identifier) @name)) @definition.class
(function_declarator declarator: (identifier) @name) @definition.function
(type_definition declarator: (type_identifier) @name) @definition.type
(enum_specifier name: (type_identifier) @name) @definition.type
```

Image:
Screenshot 2023-12-15 at 1 45 57 PM

File: website/docs/programming-languages.md
Comment (wsxiaoys) [inline website/docs/programming-languages.md:37]: Add C/C++ to `Supported Languages`, and mentioned it will be supported since 0.8.0

**Review by wsxiaoys — APPROVED** [review]
Otherwise LGTM, thanks for the effort!

**Comment by Squadrick** [comment]
@wsxiaoys Done. Also squashed the commits since I had to change the author to my personal email. Should be good to go now. Thanks for the review.

**Pull request merged**
